code/modules/mob/living/blood.dm

```diff
@@ -25,39 +25,80 @@
 			adjust_nutrition(-nutrition_ratio * HUNGER_FACTOR * seconds_per_tick)
 			blood_volume = min(blood_volume + (BLOOD_REGEN_FACTOR * nutrition_ratio * seconds_per_tick), BLOOD_VOLUME_NORMAL)
 
+	// Some effects are halved mid-combat.
+	var/determined_mod = has_status_effect(/datum/status_effect/determined) ? 0.5 : 0
+
 	//Effects of bloodloss
-	if(!(sigreturn & HANDLE_BLOOD_NO_EFFECTS))
-		var/word = pick("dizzy","woozy","faint")
-		switch(blood_volume)
-			if(BLOOD_VOLUME_MAX_LETHAL to INFINITY)
-				if(SPT_PROB(7.5, seconds_per_tick))
-					to_chat(src, span_userdanger("Blood starts to tear your skin apart. You're going to burst!"))
-					investigate_log("has been gibbed by having too much blood.", INVESTIGATE_DEATHS)
-					inflate_gib()
-			if(BLOOD_VOLUME_EXCESS to BLOOD_VOLUME_MAX_LETHAL)
-				if(SPT_PROB(5, seconds_per_tick))
-					to_chat(src, span_warning("You feel your skin swelling."))
-			if(BLOOD_VOLUME_MAXIMUM to BLOOD_VOLUME_EXCESS)
-				if(SPT_PROB(5, seconds_per_tick))
-					to_chat(src, span_warning("You feel terribly bloated."))
-			if(BLOOD_VOLUME_OKAY to BLOOD_VOLUME_SAFE)
-				if(SPT_PROB(2.5, seconds_per_tick))
-					to_chat(src, span_warning("You feel [word]."))
-				adjustOxyLoss(round(0.005 * (BLOOD_VOLUME_NORMAL - blood_volume) * seconds_per_tick, 1))
-			if(BLOOD_VOLUME_BAD to BLOOD_VOLUME_OKAY)
-				adjustOxyLoss(round(0.01 * (BLOOD_VOLUME_NORMAL - blood_volume) * seconds_per_tick, 1))
-				if(SPT_PROB(2.5, seconds_per_tick))
-					set_eye_blur_if_lower(12 SECONDS)
-					to_chat(src, span_warning("You feel very [word]."))
-			if(BLOOD_VOLUME_SURVIVE to BLOOD_VOLUME_BAD)
-				adjustOxyLoss(2.5 * seconds_per_tick)
-				if(SPT_PROB(7.5, seconds_per_tick))
-					Unconscious(rand(20,60))
-					to_chat(src, span_warning("You feel extremely [word]."))
-			if(-INFINITY to BLOOD_VOLUME_SURVIVE)
-				if(!HAS_TRAIT(src, TRAIT_NODEATH))
-					investigate_log("has died of bloodloss.", INVESTIGATE_DEATHS)
-					death()
+	if(sigreturn & HANDLE_BLOOD_NO_EFFECTS)
+		return
```

Comment:
You can't early return this there's code that runs after

Reply:
bro what thats the signal to prevent said code from running?

Reply:
checked and there's nothing below this code that needs to be ran

Reply:
look again, we never early returned to stop this from running:

```dm
	var/temp_bleed = 0
	//Bleeding out
	for(var/obj/item/bodypart/iter_part as anything in bodyparts)
		var/iter_bleed_rate = iter_part.get_modified_bleed_rate()
		temp_bleed += iter_bleed_rate * seconds_per_tick
		if(iter_part.generic_bleedstacks) // If you don't have any bleedstacks, don't try and heal them
			iter_part.adjustBleedStacks(-1, 0)
	if(temp_bleed)
		bleed(temp_bleed)
		bleed_warn(temp_bleed)
```

maybe split this out into it's own proc and run it in the sigreturn early return and at the end of this proc? unsure